plenoptic/metric/perceptual_distance.py

```diff
@@ -56,9 +57,9 @@ def _ssim_parts(img1, img2, dynamic_range, pad=False):
                           f" img1: {img_ranges[0]}, img2: {img_ranges[1]}. "
                           "Continuing anyway...")
     (n_batches, n_channels, height, width) = img1.shape
```

Comment:
Should probably wrap this in a try/except so that it gives a helpful error message if `img1` is not 4d (currently, this returns `ValueError: not enough values to unpack (expected 4, got X)`)

In general, we should probably write a helper function somewhere that checks dimensionality of images, because we'll want to use it in a lot of places.

And also explicitly check that `img2` is 4d as well.

Reply:
True, most functions need to check the dimensionality of images. I think for this PR I will not do a systematic fix.